plugins/source/aws/resources/services/lambda/functions_fetch.go

```diff
@@ -47,6 +47,11 @@ func getFunction(ctx context.Context, meta schema.ClientMeta, resource *schema.R
 			}
 			return nil
 		}
+		if client.IsAWSError(err, "AccessDenied") {
```

Comment:
Why not put it as an `or` in L45?

Reply:
I was specifically leaving that out as we want to be clear that we are missing some information because of a permission issue

Reply:
Oh ok, so maybe a comment? I checked the sdk and apparently you're right and the resource will stay even in case of an error from the PreResourceResolver (albeit half baked)

Reply:
Added a comment in the code